### DISCUSSION TITLE
HookConsumerWidget prevents Provider's context.select() widget update

### DISCUSSION THREAD

**Discussion by tomekit** [rrousselGit/riverpod#3003] [Q&A]
Using `2.4.3` version of Riverpod.

Got: `HookConsumerWidget` widget A, because I need: `ref` in the `build`. One of the child widgets of A uses.
```
final name = context.select((SomeState s) => s.name);
```
in the `build` method, which is Provider's construct: https://pub.dev/packages/provider

The issue is that after `name` is mutated in the `SomeState` my widget doesn't update.

As long as I replace: `HookConsumerWidget` in the widget A to `StatelessWidget` the widget update works fine, however I can't use `StatelessWidget`, since I need `ref` to get some dependencies, e.g.: ` ref.watch(someProvider.notifier)`

Is this a bug or some expected limitation of `riverpod`? In which case how do I workaround it?

Thanks !

**Comment by rrousselGit** [comment]
I'd need a complete example to play around with it. As it is, it's hard to say whether there's a bug or a mistake on your side.
